controllers/firewall_controller.go

```diff
@@ -283,6 +291,52 @@ func (r *FirewallReconciler) reconcileRules(ctx context.Context, f firewallv1.Fi
 	return nil
 }
 
+func (r *FirewallReconciler) reconcileEveboxAgent(ctx context.Context, f firewallv1.Firewall, log logr.Logger) error {
+	if f.Spec.IDS == nil {
+		return nil
+	}
+
+	var secrets corev1.SecretList
+	if err := r.List(ctx, &secrets, &client.ListOptions{Namespace: namespace}); err != nil {
+		// we'll ignore not-found errors, since they can't be fixed by an immediate
+		// requeue (we'll need to wait for a new notification), and we can get them
+		// on deleted requests.
+		return client.IgnoreNotFound(err)
+	}
```

Comment:
the secret should only be looked up if basic auth is enabled otherwise this is an early return.

Reply:
Done